Add mode parameter for docker sources.list file

gardouille opened the pull request:
The default value from the module (420) might block some users from
using apt* commands (such as monitoring tools, standard users,…).

ypid committed:
[debops.cryptsetup] Support an macOS Ansible controller. Fixes #31.

It has been checked that `head --byte` behaves like `iflag=fullblock`.

> "accumulate full blocks of input" is of crucial importance here as explained in [Create random data with dd and get “partial read warning”. Is the data after the warning now really random?](https://unix.stackexchange.com/questions/121865/create-random-data-with-dd-and-get-partial-read-warning-is-the-data-after-the). Without the `iflag=fullblock` and because of the blocking nature of `/dev/random` it would be very likely that `dd` only reads partial blocks which will result in a much smaller key size than what we specified. In the worst case, dd would only return one byte or maybe even none at all which would then be used as your key. I guess I don’t need to tell you how easy that would be to brute force. We can not allow this to happen.

ypid committed:
[debops.cryptsetup] There is no `head --byte` or `head --bytes` on macOS

https://github.com/debops/debops/pull/232#issuecomment-374000556

There's no `--byte` option on BSD/Apple `head`.
`head -c X` is the right command for it -- see [manpage](http://www.manpagez.com/man/1/head/)

Also, from what I see even GNU head doesn't have "--byte" option advised in man/help (however, it works, so I guess it's just left for backward compatibility)

Both GNU and BSD head support `-c` option, and it is in fact equivalent to `--byte` afaiu:

```
❯ ghead --help|grep -- --byte
  -c, --bytes=[-]NUM       print the first NUM bytes of each file;
```

And I assume that `--byte` is same as `-c`:

```
❯ ghead --byte
ghead: option '--bytes' requires an argument
Try 'ghead --help' for more information.
```

Thanks to: @stek29

ypid committed:
[debops-contrib.tor] Automatic firewall config, Fix auto update of tor

* Add automatic firewall config based on tor_ports
* Fix auto update of tor

This is still work in progress but it should be usable.

ypid committed:
[debops.elastic_co] Elasticsearch BV. renamed to Elastic in 2015

Ref: https://www.elastic.co/about/press/elasticsearch-changes-name-to-elastic-to-reflect-wide-adoption-beyond-search

ypid committed:
[debops.elastic_co] Install Elastic products of version 6.x by default

ypid committed:
[docs] Elastic prefers the term "Elastic Stack" over "ELK Stack"

Ref: https://www.elastic.co/elk-stack
Ref: https://changelog.com/podcast/292

ypid committed:
[debops.kibana] Installation of plugins should be done as kibana__user

Ref: https://www.elastic.co/guide/en/kibana/6.2/installing-xpack-kb.html
Ref: https://github.com/elastic/kibana/issues/8818

ypid committed:
[debops.elasticsearch] Fix logging of Elasticsearch

It seems Elasticsearch 6.2 needs to have `path.logs` configured in order
to output useful logs.

ypid committed:
[debops.java] Test that the role works on Debian Stretch

ypid committed:
[debops.elasticsearch] Test that the role works on Debian Stretch

ypid committed:
[debops.java] Remove useless shell=True from Ansible facts python script

ypid committed:
Use the command Ansible module instead of shell where possible

ypid committed:
[debops.nginx] Fix undefined variable issue with item.auth_basic_name

Seems Ansible 2.5 has issues when item.auth_basic_name is not defined
but not used because item.auth_basic_filename is set.

AnsibleUndefinedVariable: 'dict object' has no attribute 'auth_basic_name'

ypid committed:
[docs] Improve docs related to the Elastic Stack and fix typos

ypid committed:
[debops.java] Coding style improvement to match DebOps Standards

gasuketsu committed:
[debops.mariadb_server] Fix failure with non-default datadir

* Stop database server on first install
* Move files under default datadir to given datadir

gasuketsu committed:
use default datadir variable instead of literal

gardouille committed:
Add mode parameter for docker sources.list file

The default value from the module (420) might block some users from
using apt* commands (such as monitoring tools, standard users,…).

gardouille committed:
Add mode parameter for docker sources.list file

The default value from the module (420) might block some users from
using apt* commands (such as monitoring tools, standard users,…).

gardouille commented:
The default mode is already overwrite for some other roles (apt_install, mariadb,…), i choose the same value (0644).

I set it in a variable to allow users to overwrite it if they really want to block this access.

In my case, i wasn't able to use apt-cache policy without sudo/su permissions ^^

drybjed committed:
Merge branch 'ypid-rework/elastic_roles'

drybjed committed:
Merge branch 'rework/debops-contrib.tor' of https://github.com/ypid/debops into ypid-rework/debops-contrib.tor

drybjed committed:
Merge branch 'ypid-rework/debops-contrib.tor'

drybjed committed:
Merge branch 'rework/mariadb-server-datadir' of https://github.com/gasuketsu/debops into gasuketsu-rework/mariadb-server-datadir

drybjed committed:
Merge branch 'gasuketsu-rework/mariadb-server-datadir'

drybjed reviewed (changes requested):
On my hosts the Docker `sources.list` file already has `0644` permissions. Maybe its due to your OS choice or Ansible version?

Anyway, I guess we can set the mode in the task, there's no issue with using `0644` there. Since APT sources are "public", I don't see any need to have it as a default variable though, you could just set it directly. I don't think that anybody would want to change the default `0644` to anything else.

drybjed committed:
[debops.nodejs] Prefer 'libssl-dev' from backports

Since the installation of backported NodeJS package requires
'libssl1.0.0' package from backports, add 'libssl-dev' APT preference
from backports as well, so that the correct version can be installed if
needed.

drybjed committed:
[debops.nodejs] Install fixed NPM version from git

Newer NPM releases work only with NodeJS 6.0.0+, therefore on older
installations, an older, specific NPM version will be installed from
git repository.

drybjed committed:
[debops.nodejs] Rename 'nodejs__upstream_version'

The variable is renamed to 'nodejs__upstream_release' to better
represent its contents, which is not a specific NodeJS version but
a major release number.

drybjed committed:
[debops.nodejs] Update upstream release comments

drybjed committed:
[nodejs] Install upstream NodeJS 8.x by default

drybjed committed:
[debops.nodejs] Install upstream NodeJS by default

The NodeJS packages released with Debian Stretch do not have security
support, therefore using upstream packages instead should be considered
more secure.

drybjed committed:
Merge branch 'drybjed-nodejs-changes'

drybjed committed:
[debops.gitlab] Redesgin version management

drybjed committed:
[debops.gitlab] Invert release selection logic

drybjed committed:
[debops.gitlab] Install 'gitaly' as 'git' account

drybjed committed:
[debops.gitlab] Add support for GitLab 10.7

drybjed committed:
Merge branch 'drybjed-gitlab-versions'

drybjed committed:
Merge branch 'fix/issue-31' of https://github.com/ypid/debops into ypid-fix/issue-31

drybjed committed:
Merge branch 'ypid-fix/issue-31'

drybjed committed:
[debops.sshd] Don't block SSH access if standalone

The 'debops.sshd' role checks the Ansible local facts of the
'debops.system_groups' role to get the list of UNIX groups which are
allowed to login and configure them in the 'AllowGroups' parameter.

This change ensures that if the above facts are not set, ie.
'debops.sshd' role is used as standalone, the 'AllowGroups' will not be
set - otherwise this would result in locked down system access where
only 'root' account can login, which might not be enabled.

drybjed committed:
Merge branch 'sshd-allow-groups' of https://github.com/drybjed/debops into drybjed-sshd-allow-groups

drybjed committed:
Merge branch 'drybjed-sshd-allow-groups'

Commit:
[debops.ferm] Handle correctly protocols for DMZ entries.

drybjed committed:
Merge branch 'blaa-master'

gardouille committed:
Merge branch 'master' of https://github.com/gardouille/debops

gardouille committed:
Fix sources.list mode in task file

gardouille commented:
Close this and create a new clean PR… :)
